### DISCUSSION TITLE
"No GPU found" issue.

### DISCUSSION THREAD

**Discussion by okliis** [acids-ircam/RAVE#110] [Q&A] [answered]
Hi all,
I keep getting this when starting training: 

"No GPU found.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs"

I have CUDA installed and am successfully using it on another PyTorch project.
Does anyone know how to deal with this? 
For context, I'm in a python 3.9.9 venv on Windows 10, all requirements installed.  

Thanks in advance,
Mike

**Comment by okliis** [comment, marked as answer]
Sorted it out. 
Ended up using conda for my virtualenv, reinstalled requirements, GPU was finally recognized.

**Reply by clusternd** [reply]
Would you explain in detail please what you have done to make this work? What is "virtualenv"?
I am also in Win 10 having the same issues and quite new to this area.

**Comment by znmeb** [comment]
The best documentation for installing PyTorch can be found here:

https://pytorch.org/get-started/locally/

There are instructions for both `conda` and `pip`. You will get a working GPU if you follow them.

If your project needs an older version of PyTorch, they can be found here:

https://pytorch.org/get-started/previous-versions/

**Comment by jacklion710** [comment]
Just curious if anyone can clarify for me. When you mentioned that you reinstalled the requirements, did you reinstall rave with your virtualenv or are you just referring to ffmpeg and pytorch? When I try and install acids-rave in my conda environment I get the following error:

```
(rave_env) C:\Users\Jake>conda install -n rave_env acids-rave
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - acids-rave

Current channels:

  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```


Since I can't install acids-rave into the conda environment I tried running the following line...
`rave preprocess --input_path C:/Users/Jake/Desktop/JLSounds --output_path C:/Users/Jake/Desktop/Dataset (--lazy)`

But it results in this error...
```
(rave_env) C:\Users\Jake>rave preprocess --input_path C:/Users/Jake/Desktop/JLSounds --output_path C:/Users/Jake/Desktop/Dataset (--lazy)
Traceback (most recent call last):
  File "C:\Users\Jake\anaconda3\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Jake\anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Jake\anaconda3\Scripts\rave.exe\__main__.py", line 7, in <module>
  File "C:\Users\Jake\anaconda3\lib\site-packages\scripts\main_cli.py", line 34, in main
    from scripts import preprocess
  File "C:\Users\Jake\anaconda3\lib\site-packages\scripts\preprocess.py", line 13, in <module>
    import torch
  File "C:\Users\Jake\anaconda3\lib\site-packages\torch\__init__.py", line 128, in <module>
    raise err
OSError: [WinError 127] The specified procedure could not be found. Error loading "C:\Users\Jake\anaconda3\lib\site-packages\torch\lib\shm.dll" or one of its dependencies.
```

**Reply by jacklion710** [reply]
Nevermind! Turned out I just needed to run the cmd shell as an admin
